core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -251,6 +243,8 @@ class SearchService(
   /** only for exact fqns */
   def findUnique(fqn: String): Option[FqnSymbol] = Await.result(db.find(fqn), QUERY_TIMEOUT)
 
+  def getClassHierarchy(fqn: String, hierarchyType: HierarchyType): Option[Hierarchy] = Await.result(db.getClassHierarchy(fqn, hierarchyType), QUERY_TIMEOUT)
```

Comment:
the legacy API demanded `Option` instead of `Future`, if it's possible we should go for async here